tests/test_unit_conversion.py

```diff
@@ -1,80 +1,1235 @@
-    with pytest.raises(AssertionError, match='^expression must be of type Sympy expression$'):
-        simple_units_model.units.get_conversion_factor(to_unit, expression=quantity)
-
-    # unit to unit
-    assert simple_units_model.units.get_conversion_factor(to_unit=to_unit, from_unit=from_unit) == 1000
-    # quantity to unit
-    assert simple_units_model.units.get_conversion_factor(to_unit=to_unit, quantity=quantity) == 1000
-    # expression to unit
-    assert simple_units_model.units.get_conversion_factor(to_unit=to_unit, expression=expression) == 1000
-
-
-def test_conversion_factor_same_units(simple_units_model):
-    """ Tests Units.get_conversion_factor() function when units are same
-    and conversion factor should be '1'. """
-    symbol_b = simple_units_model.get_symbol_by_cmeta_id("b")
-    equation = simple_units_model.get_equations_for([symbol_b])
-    expression = equation[1].rhs
-    to_unit = simple_units_model.units.ureg('per_ms').units
-    from_unit = simple_units_model.units.summarise_units(expression)
-    quantity = 1 * from_unit
-    # quantity to unit
-    assert simple_units_model.units.get_conversion_factor(to_unit=to_unit, quantity=quantity) == 1
-    # unit to unit
-    assert simple_units_model.units.get_conversion_factor(to_unit=to_unit, from_unit=from_unit) == 1
-    # expression to unit
-    assert simple_units_model.units.get_conversion_factor(to_unit=to_unit, expression=expression) == 1
+class TestUnitConversion:
+    ###############################################################
+    # fixtures
+
+    @pytest.fixture
+    def local_model(scope='function'):
+        """ Fixture to load a local copy of  the basic_ode model that may get modified. """
+        return shared.load_model('basic_ode')
+
+    @pytest.fixture
+    def model_missing_units(scope='function'):
+        """ Fixture to load a local copy of  the basic_ode model that may get modified. """
+        return shared.load_model('missing_units_for_conversion_tests')
+
+    @pytest.fixture
+    def literals_model(scope='function'):
+        """ Fixture to load a local copy of  the basic_ode model that may get modified. """
+        return shared.load_model('literals_for_conversion_tests')
+
+    @pytest.fixture
+    def multiode_model(scope='function'):
+        """ Fixture to load a local copy of  the basic_ode model that may get modified. """
+        return shared.load_model('repeated_ode_for_conversion_tests')
+
+    @pytest.fixture
+    def multiode_freevar_model(scope='function'):
+        """ Fixture to load a local copy of  the basic_ode model that may get modified. """
+        return shared.load_model('repeated_ode_freevar_for_conversion_tests')
+
+    @pytest.fixture
+    def silly_names(scope='function'):
+        """ Fixture to load a local copy of  the basic_ode model that may get modified. """
+        return shared.load_model('silly_names')
+
+    def test_add_preferred_custom_unit_name(self, simple_ode_model):
```

Comment:
This isn't really a unit conversion test - should it be outside this class still?